cmd/kubeadm/app/preflight/checks.go

```diff
@@ -415,19 +416,20 @@ func (hst HTTPProxyCheck) Name() string {
 // Check validates http connectivity type, direct or via proxy.
 func (hst HTTPProxyCheck) Check() (warnings, errors []error) {
 
-	url := fmt.Sprintf("%s://%s:%d", hst.Proto, hst.Host, hst.Port)
+	hostPort := net.JoinHostPort(hst.Host, strconv.Itoa(hst.Port))
+	u := (&url.URL{Scheme: hst.Proto, Host: hostPort}).String()
 
-	req, err := http.NewRequest("GET", url, nil)
+	req, err := http.NewRequest("GET", u, nil)
 	if err != nil {
 		return nil, []error{err}
 	}
 
-	proxy, err := http.DefaultTransport.(*http.Transport).Proxy(req)
+	proxy, err := netutil.SetOldTransportDefaults(&http.Transport{}).Proxy(req)
```

Comment:
Interesting. this line should actually trigger merge conflicts, as it was already changed in 4bd692a3bff

Reply:
I'll rebase to pick up those changes.